CMakeLists.txt

```diff
@@ -158,23 +158,19 @@ endforeach()
 
 include(ExternalProject)
 
-
-# CMAKE_BUILD_WITH_INSTALL_RPATH should be set to true.
-# It is because `nekotools boot` will use the neko in build dir during build.
-set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-
 if (RELOCATABLE)
-	# https://cmake.org/Wiki/CMake_RPATH_handling
-	set(CMAKE_SKIP_BUILD_RPATH FALSE)
-	set(CMAKE_SKIP_INSTALL_RPATH FALSE)
+	# https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/RPATH-handling
+	# Set this to true, otherwise the binaries won't be relocatable until after installation:
+	set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 	if(APPLE)
-		set(CMAKE_MACOSX_RPATH TRUE)
 		set(CMAKE_INSTALL_RPATH @executable_path/)
 	elseif(UNIX)
 		set(CMAKE_INSTALL_RPATH \$ORIGIN)
 	endif()
 endif()
 
+list(APPEND CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_FULL_LIBDIR})
```

Comment:
@tobil4sk 
Debian packages do not want any custom rpath on their binary. Can this be put behind a conditional such that I can turn it off somehow?

Reply:
I've just commited https://github.com/HaxeFoundation/neko/commit/9638b62fd42fcfd4c21499b6531dccf7d35f82ae

Reply:
We need this on linux distros as well when building from source, not just on macOS. What about this flag? [`CMAKE_SKIP_INSTALL_RPATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_SKIP_INSTALL_RPATH.html)? Maybe it would work without any changes to the cmake script?

Reply:
@andyli Could you see if this works? https://github.com/HaxeFoundation/neko-debian/pull/4

Reply:
> We need this on linux distros as well when building from source, not just on macOS.

How did you build & install?

I can build, install neko in /usr/local, and run it without issue.
Commands I used:
```
git clone https://github.com/HaxeFoundation/neko.git

docker run -it --rm -v $(pwd)/neko:/neko ubuntu:jammy bash

apt update
apt install -y build-essential cmake git pkg-config libgc-dev libssl-dev libpcre2-dev zlib1g-dev apache2-dev libmariadb-client-lgpl-dev-compat libsqlite3-dev libmbedtls-dev libgtk-3-dev

cd /neko
mkdir -p build
cd build
cmake ..
make
make install

which neko
neko -version
neko bin/test.n
```

Reply:
I've built and installed using the equivalent commands on Arch linux (no docker though). This issue is not exclusive to Arch, most distributions don't use `/usr/local/lib` as a default loader path which is why it fails to load:
```
$ neko
neko: error while loading shared libraries: libneko.so.2: cannot open shared object file: No such file or directory
```
If I run `$ sudo ldconfig /usr/local/lib` then it works until I reboot my machine, or setting `LD_LIBRARY_PATH=/usr/local/lib` also works.

Since the default install location is `/usr/local/`, then the rpath should be set by default otherwise the default build is broken for users building from source on most distros. If https://github.com/HaxeFoundation/neko-debian/pull/4 does not work, then we can add our own flag to disable it.

Reply:
https://github.com/HaxeFoundation/neko-debian/pull/4 works and I'm gonna use it.

I still feel like leaving CMAKE_INSTALL_RPATH unset is a better default, but I will revert 9638b62fd42fcfd4c21499b6531dccf7d35f82ae. Neko 2.4.0 was released anyway so I guess it's better to keep it that way.

Reply:
> https://github.com/HaxeFoundation/neko-debian/pull/4 works and I'm gonna use it.

Thanks for trying the patch and merging.

> I still feel like leaving CMAKE_INSTALL_RPATH unset is a better default

It is not just 2.4.0 where a flag has been necessary to disable install rpath. In the past, it was already necessary to disable `RELOCATABLE` to avoid any install RPATH being set.